go/common/grpc/grpc.go

```diff
@@ -473,11 +474,14 @@ func NewServer(config *ServerConfig) (*Server, error) {
 	sOpts = append(sOpts, grpc.CustomCodec(&CBORCodec{}))
 	sOpts = append(sOpts, config.CustomOptions...)
 
-	if config.Certificate != nil {
+	if config.Identity != nil {
 		tlsConfig := &tls.Config{
-			Certificates: []tls.Certificate{*config.Certificate},
-			ClientAuth:   clientAuthType,
+			ClientAuth: clientAuthType,
+			GetCertificate: func(ch *tls.ClientHelloInfo) (*tls.Certificate, error) {
+				return config.Identity.TLSCertificate, nil
```

Comment:
This seems racy. Can you change it so that getting the `TLSCertificate` from identity is synchronized with rotation (e.g., add accessor that uses a mutex)?